src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -279,13 +281,50 @@ internal void PublishRpcChannelReadyEvent(RpcEvent initEvent)
                 return;
             }
 
-            RpcChannelReadyEvent readyEvent = new RpcChannelReadyEvent(_workerId, _workerConfig.Language, this, _initMessage.WorkerVersion, _initMessage.Capabilities);
-            _eventManager.Publish(readyEvent);
+            var managedDepedencyProvider = new ManagedDependencyProvider();
+            var osPlatform = OSPlatform.Windows;
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                osPlatform = OSPlatform.Linux;
+            }
 
-            if (_functionRegistrations != null)
+            var managedDependencies = managedDepedencyProvider.GetDependencies(Path.Combine(_rootScriptPath, ScriptConstants.HostMetadataFileName), osPlatform, new ManagedDependencyConfigs()).GetAwaiter().GetResult();
+            if (_initMessage.Capabilities.ContainsKey("ManagedDependenciesEnabled")
+                && string.Equals(_initMessage.Capabilities["ManagedDependenciesEnabled"], "true", StringComparison.InvariantCultureIgnoreCase)
```

Comment:
Please add this to .proto as a enum.  Language worker can reuse this enum to publish capabilities